### PR TITLE
Ensure consistent link underline and color on hover 

### PR DESCRIPTION
**Pull request by s596757**
Made CSS changes in order to ensure all links are visually distinct:

All links are now underlined as standard expectation based on:
https://www.w3.org/TR/html401/struct/links.html
When you hover over a link, the link is both bold and underlined
Link that has been visited changes to a grey color
Tested visually
Based on feedback from UX engineer

Ticket based on issue https://github.com/jupyterlab/jupyterlab/issues/9399 Text issue number 2

Please provide additional feedback and notes on your definition of visually distinct

### CODE REVIEW

File: packages/rendermime/style/base.css [hunk 1]
```diff
@@ -32,18 +32,19 @@
...
   text-decoration: underline;
   color: var(--jp-content-link-color);
+  font-weight: bold;
 }
 
 .jp-RenderedText pre a:visited {
-  text-decoration: none;
-  color: var(--jp-content-link-color);
+  text-decoration: underline;
+  color: var(--jp-content-font-color3);
```
Comment (krassowski) [inline packages/rendermime/style/base.css:47]: Can you define a semantic variable, e.g. `--jp-content-link-visited-color` or `--jp-content-visited-link-color` and use it where needed to allow overriding the colour choice and ensure consistency?

```suggestion
  color: var(--jp-content-link-visited-color);
```

**Review by krassowski — CHANGES_REQUESTED** [review]
> All links are now underlined as standard expectation based on: https://www.w3.org/TR/html401/struct/links.html

This is not what the document linked to says - it says *for example* underlined

> When you hover over a link, the link is both bold and underlined

I do not agree with this design direction. All of the web uses underline on hover. Using bold on hover will lead to layout shifts. So this is -1 from me here.

> Link that has been visited changes to a grey color

Changing colour to indicate visited links is probably fine, but I would request to add a semantic variable so that themes (and users) can override it easily.

File: packages/rendermime/style/base.css [hunk 2]
```diff
@@ -256,18 +257,19 @@
...
   text-decoration: underline;
   color: var(--jp-content-link-color);
+  font-weight: bold;
 }
 
 .jp-RenderedHTMLCommon a:visited {
-  text-decoration: none;
-  color: var(--jp-content-link-color);
+  text-decoration: underline;
+  color: var(--jp-content-font-color3);
```
Comment (krassowski) [inline packages/rendermime/style/base.css:272]: Should use `var(--jp-content-link-visited-color)` too.

File: packages/extensionmanager/style/base.css [hunk 3]
```diff
@@ -188,8 +190,8 @@
 }
 
 .jp-extensionmanager-entry-name a:visited {
-  color: var(--jp-content-link-color);
-  text-decoration: none;
+  color: var(--jp-content-font-color3);
```
Comment (krassowski) [inline packages/extensionmanager/style/base.css:193]: Should use `var(--jp-content-link-visited-color)` too.

File: packages/rendermime/style/base.css [hunk 4]
```diff
@@ -32,18 +32,19 @@
...
   text-decoration: underline;
   color: var(--jp-content-link-color);
+  font-weight: bold;
 }
 
 .jp-RenderedText pre a:visited {
-  text-decoration: none;
-  color: var(--jp-content-link-color);
+  text-decoration: underline;
+  color: var(--jp-content-link-visited-color);
```
Comment (krassowski) [inline packages/rendermime/style/base.css:47]: I do not see a definition for this variable, it needs to be added.

**Review by krassowski — CHANGES_REQUESTED** [review]
My previous opinion that we should not switch to underline/bold still stands. However, I do think that maybe it could be opt-in via a setting (which would involve creating a dedicated accessibility package).

**Comment by krassowski** [comment]
@s596757 the lint job is failing can you run `jlpm run prettier`?

**Comment by s596757** [comment]
> jlpm run prettier

I have run it :)

**Comment by krassowski** [comment]
@s596757 did you commit and push changes?  The [lint check](https://github.com/jupyterlab/jupyterlab/actions/runs/5819136652/job/15776958467?pr=14914) is still failing with:

```
Checking formatting...
[warn] packages/rendermime/style/base.css
[warn] packages/theme-dark-extension/style/variables.css
[warn] Code style issues found in 2 files. Run Prettier to fix.
+ echo 'Please run `jlpm run prettier` locally and push changes'
Please run `jlpm run prettier` locally and push changes
```

File: packages/theme-dark-extension/style/variables.css [hunk 5]
```diff
@@ -167,6 +167,8 @@ all of MD as it is not optimized for dense, information rich UIs.
   --jp-content-font-color2: rgba(255, 255, 255, 0.7);
   --jp-content-font-color3: rgba(255, 255, 255, 0.5);
   --jp-content-link-color: var(--md-blue-300);
+  --jp-content-link-visited-color: rgba(255,255,255, 0.3);
+  --jp-content-link-hover-color: var(--md-blue-400)
```
Comment (krassowski) [inline packages/theme-dark-extension/style/variables.css:171]: Educated guess that this is why the lint check is failing.

```suggestion
  --jp-content-link-hover-color: var(--md-blue-400);
```

**Comment by krassowski** [comment]
@s596757 I addressed the lint failure for you, can you add before/after screenshots in both light and dark mode to the top-level comment, please?

**Comment by krassowski** [comment]
Closing in favour of #15181.

**Pull request closed without merging**